packages/app-metadata-controller/src/AppMetadataController.test.ts

```diff
@@ -0,0 +1,165 @@
+import { Messenger } from '@metamask/base-controller';
+
+import {
+  AppMetadataController,
+  getDefaultAppMetadataControllerState,
+  type AppMetadataControllerOptions,
+} from './AppMetadataController';
+
+describe('AppMetadataController', () => {
+  describe('constructor', () => {
+    it('accepts initial state and does not modify it if currentMigrationVersion and platform.getVersion() match respective values in state', async () => {
+      const initState = {
+        currentAppVersion: '1',
+        previousAppVersion: '1',
+        previousMigrationVersion: 1,
+        currentMigrationVersion: 1,
+      };
+      withController(
+        {
+          state: initState,
+          currentMigrationVersion: 1,
+          currentAppVersion: '1',
+        },
+        ({ controller }) => {
+          expect(controller.state).toStrictEqual(initState);
+        },
+      );
+    });
+
+    it('sets default state and does not modify it', () => {
+      withController({ state: {} }, ({ controller }) => {
```

Comment:
It seems like this test is designed to exercise the scenario where the controller initializes itself with default state, in which case we don't need to pass any state. If we remove that option here that should satisfy CI.

``` suggestion
      withController(({ controller }) => {
```